lib/internal/http2/core.js

```diff
@@ -2841,6 +2852,33 @@ class Http2Server extends NETServer {
   }
 }
 
+Http2Server.prototype[EventEmitter.captureRejectionSymbol] = function(
+  err, event, ...args) {
+
+  switch (event) {
+    case 'stream':
+      const [stream] = args;
+      stream.destroy(err);
+      break;
```

Comment:
Shouldn’t this also result in a 500 if nothing has been sent? I’m not 100% on how that would work but could probably translate from the compat case.

Reply:
@jasnell what do you think?

Reply:
Yes, it should, but only if a response has not yet been sent

Reply:
Done